test/functional/testrunner.js

```diff
@@ -51,29 +51,115 @@ createReport = () => {
   );
 };
 
-createTestCafe().then(tc => {
-  testcafe = tc;
-  const runner = testcafe.createRunner();
-  runSuite = suite => {
-    const runOptions = {
-      skipJsErrors: config.desktop.skipCriticalConsoleJsErrors,
-      quarantineMode: config.desktop.quarantineMode,
-      speed: config.desktop.speed,
-      debugMode: false,
-      selectorTimeout: config.desktop.selectorTimeOut,
-      assertionTimeout: config.desktop.assertionTimeout
+const isSL = process.argv.includes("--sl");
+if (environment === "dev" && isSL === false) {
+  createTestCafe().then(tc => {
```

Comment:
I don't understand what is different in each environment.  Why the 4 cases when they all do the same thing?